### ISSUE TITLE
Automate jobs with the usage of Travis CI

### ISSUE THREAD

**Issue by JonatanSalas** [redux-autoform/redux-autoform-material-ui#40]
@danigomez  @andrerpena  When performing a build of every commit to keep some parts of the project sync, we should be able to do:
- Automate **gh-pages** deployment
- Automate **npm** distribution based on changes detected over **package.json** 

In this way we could work better. I will charge with this task, to configure all the projects that has a **travis.yml** like **redux-autoform** and **redux-autoform-bootstrap-ui** 

What's your opinion?

**Comment by andrerpena** [comment]
Are you telling me that it's possible to automate Travis to publish to gh-pages automatically? I mean.. really? I know it's possible and I think it makes sense to do it as an NPM script but I didn't know this could be delegated to Travis.

**Comment by JonatanSalas** [comment]
You could create a script in NPM, and after a success build tell to Travis-CI that executes it. In this way, we keep the demo up to date, and when build fails, it keeps the demo that doesn't have any trouble!

**Comment by andrerpena** [comment]
Yes but if Travis does it, it'd need Git credentials to push it to GH. And we woudn't commit Git credentials right? Or does it have a magical way that I'm not aware of?

**Comment by JonatanSalas** [comment]
When you enable travis for your repo, it directly interacts with the Github API so, if you give permissions to Travis, then, it could do this. I'm interested in this plugin, and that's why I want to add gulp! https://github.com/shinnn/gulp-gh-pages

We could make tasks with gulp and tell travis that execute it.

**Comment by andrerpena** [comment]
Oh god I'm so behind the time. Ok, that's amazing. If you could do it I'd really appreciate.

**Comment by JonatanSalas** [comment]
I will work on this so we can have a good dev enviroment :smile:

**Comment by JonatanSalas** [comment]
@andrerpena waiting for this! :disappointed_relieved:  If the creator not respond to the issue, then I will try to add and see if it pass with travis! :open_mouth:  

https://github.com/shinnn/gulp-gh-pages/issues/100
